docs/src/examples.md

```diff
@@ -0,0 +1,609 @@
+data = """
+col1,col2
+"quoted field with a delimiter , inside","quoted field that contains a \\n newline and ""inner quotes\"\"\"
+unquoted field,unquoted field with "inner quotes"
+"""
+
+file = CSV.File(file; quotechar='"', escapechar='"')
+
+file = CSV.File(file; openquotechar='"' closequotechar='"', escapechar='"')
+```
+
+
+### [DateFormat](@id dateformat_example)
+
+```julia
+using CSV
+
+# In this file, our `date` column has dates that are formatted like `yyyy/mm/dd`. We can pass just such a string to the `dateformat` keyword argument to tell parsing to use it when looking for `Date` or `DateTime` columns. Note that currently, only a single `dateformat` string can be passed to parsing, meaning multiple columns with different date formats cannot all be parsed as `Date`/`DateTime`.
+data = """
+code,date
+0,2019/01/01
+1,2019/01/02
+"""
+
+file = CSV.File(file; dateformat="yyyy/mm/dd")
+```
+
+### [Custom decimal separator](@id decimal_example)
+
+```julia
+using CSV
+
+# In many places in the world, floating point number decimals are separated with a comma instead of a period (`3,14` vs. `3.14`). We can correctly parse these numbers by passing in the `decimal=','` keyword argument. Note that we probably need to explicitly pass `delim=';'` in this case, since the parser will probably think that it detected `','` as the delimiter.
+data = """
+col1;col2;col3
+1,01;2,02;3,03
+4,04;5,05;6,06
+"""
+
+file = CSV.File(file; delim=';', decimal=',')
+```
+
+### [Custom bool strings](@id truestrings_example)
+
+```julia
+using CSV
+
+# By default, parsing only considers the string values `true` and `false` as valid `Bool` values. To consider alternative values, we can pass a `Vector{String}` to the `truestrings` and `falsestrings` keyword arguments.
+data = """
+id,paid,attended
+0,T,TRUE
+1,F,TRUE
+2,T,FALSE
+3,F,FALSE
+"""
+
+file = CSV.File(file; truestrings=["T", "TRUE"], falsestrings=["F", "FALSE"])
+```
+
+### [Matrix-like Data](@id matrix_example)
+
+```julia
+using CSV
+
+# This file contains a 3x3 identity matrix of `Float64`. By default, parsing will detect the delimiter and type, but we can also explicitly pass `delim= ' '` and `types=Float64`, which tells parsing to explicitly treat each column as `Float64`, without having to guess the type on its own.
```

Comment:
probably `ignorerepeated` should be also added here, as this typically happens in matrix data:
```
julia> rand(5,5)
5×5 Matrix{Float64}:
 0.885415   0.436992  0.966233  0.415127   0.595875
 0.0674271  0.586727  0.992303  0.236927   0.997092
 0.729953   0.720313  0.70169   0.922141   0.377983
 0.932993   0.257281  0.92218   0.0864256  0.0153523
 0.852068   0.591985  0.577147  0.966295   0.220887
```